src/SerializerFactory.php

```diff
@@ -41,26 +43,41 @@ class SerializerFactory {
 	const OPTION_SERIALIZE_REFERENCE_SNAKS_WITHOUT_HASH = 8;
 
 	/**
-	 * @var int
+	 * @var DispatchableSerializer[]|callable[]
 	 */
-	private $options;
+	private $entitySerializers;
 
 	/**
 	 * @var Serializer
 	 */
 	private $dataValueSerializer;
 
 	/**
+	 * @var int
+	 */
+	private $options;
+
+	/**
+	 * @since 3.0
+	 *
+	 * @param DispatchableSerializer[]|callable[] $entitySerializers A list of either
+	 *  DispatchableSerializer objects that are able to serialize Entities, or callbacks that return
+	 *  such objects.
 	 * @param Serializer $dataValueSerializer serializer for DataValue objects
 	 * @param int $options set multiple with bitwise or
 	 *
 	 * @throws InvalidArgumentException
 	 */
-	public function __construct( Serializer $dataValueSerializer, $options = 0 ) {
+	public function __construct(
+		array $entitySerializers,
+		Serializer $dataValueSerializer,
+		$options = 0
```

Comment:
As suggested in the other patch, this could also default to self::OPTION_DEFAULT so that users don't have to wonder what is this zero about.